presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -164,13 +165,16 @@ public Response createQuery(
     {
         assertRequest(!isNullOrEmpty(statement), "SQL statement is empty");
 
-        Session session = createSessionForRequest(servletRequest, accessControl, sessionPropertyManager, queryIdGenerator.createNextQueryId());
-
-        ExchangeClient exchangeClient = exchangeClientSupplier.get(deltaMemoryInBytes -> { });
-        Query query = new Query(session, statement, queryManager, exchangeClient);
-        queries.put(query.getQueryId(), query);
-
-        return getQueryResults(query, Optional.empty(), uriInfo, new Duration(1, MILLISECONDS));
+        try {
+            Session session = createSessionForRequest(servletRequest, accessControl, sessionPropertyManager, queryIdGenerator.createNextQueryId());
+            ExchangeClient exchangeClient = exchangeClientSupplier.get(deltaMemoryInBytes -> { });
+            Query query = new Query(session, statement, queryManager, exchangeClient);
+            queries.put(query.getQueryId(), query);
+            return getQueryResults(query, Optional.empty(), uriInfo, new Duration(1, MILLISECONDS));
+        }
+        catch (PrestoException e) {
```

Comment:
Let's only wrap creating the session, since the other stuff shouldn't throw `PrestoException`